app/jinja_filters.py

```diff
@@ -250,18 +250,20 @@ def get_min_max_value_width(
     """
     This function gets the minimum and maximum value accepted for a question.
     Which then allows us to use that value to set the width of the textbox to suit that min and max.
+
+    If the min or max for the answer is a value source but not an "answers" source, such as a calculated or grand calculated summary,
+    use the length of the default value for the min and max width, as the actual min and max width cannot currently be determined
     """
 
-    if (
-        answer.get(min_max, {})
-        and isinstance(answer[min_max]["value"], Mapping)
-        and answer[min_max]["value"].get("source") == "answers"
-    ):
-        schema: QuestionnaireSchema = g.get("schema")
-        identifier = answer[min_max]["value"].get("identifier")
-        return schema.min_and_max_map[identifier][min_max]
+    if answer.get(min_max, {}) and isinstance(answer[min_max]["value"], Mapping):
+        if answer[min_max]["value"].get("source") == "answers":
+            schema: QuestionnaireSchema = g.get("schema")
+            identifier = answer[min_max]["value"].get("identifier")
+            return schema.min_and_max_map[identifier][min_max]
+        return len(str(default_value))
 
-    return len(str(answer.get(min_max, {}).get("value", default_value)))
+    # Factor out the decimals as it's accounted for in get_width_for_number
+    return len(str(int(answer.get(min_max, {}).get("value", default_value))))
```

Comment:
I would do the int conversion first with the new var suggested above, as don't find this line very readable at the moment with multiple casts.

Reply:
I've added in some changes, let me know if these need to be changed to improve readability 👍